From 0c4e2ee5ae8f8ae4304787c90bfd3d00d0258d5b Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:13 -0400
Subject: [virt] kvm/nvmx: Report 2MB EPT pages as supported

Message-id: <1406040016-3289-99-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86177
O-Subject: [RHEL7 PATCH v2 098/221] KVM: nVMX: Report 2MB EPT pages as supported
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

As long as the hardware provides us 2MB EPT pages, we can also expose
them to the guest because our shadow EPT code already supports this
feature.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit d3134dbf2046076b6e52b05890e812442d5257fb)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 1056017..c77c980 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2261,7 +2261,8 @@ static __init void nested_vmx_setup_ctls_msrs(void)
   /* nested EPT: emulate EPT also to L1 */
   nested_vmx_secondary_ctls_high |= SECONDARY_EXEC_ENABLE_EPT;
   nested_vmx_ept_caps = VMX_EPT_PAGE_WALK_4_BIT |
-    VMX_EPTP_WB_BIT | VMX_EPT_INVEPT_BIT;
+    VMX_EPTP_WB_BIT | VMX_EPT_2MB_PAGE_BIT |
+    VMX_EPT_INVEPT_BIT;
   nested_vmx_ept_caps &= vmx_capability.ept;
   /*
    * Since invept is completely emulated we support both global
-- 
1.7.1